lib/engine/locations.ex

```diff
@@ -142,11 +142,11 @@ defmodule Engine.Locations do
       trip_id: location["vehicle"]["trip"]["trip_id"],
       consist: location["vehicle"]["vehicle"]["consist"],
       multi_carriage_details:
-        location["vehicle"]["multi_carriage_details"] &&
-          Enum.map(
-            location["vehicle"]["multi_carriage_details"],
-            &parse_carriage_details/1
-          )
+        (location["vehicle"]["multi_carriage_details"] &&
+           Enum.map(
+             location["vehicle"]["multi_carriage_details"],
```

Comment:
nit: If we do `location["vehicle"]["multi_carriage_details"] || []`, we could remove the outer guard and map over it unconditionally, which might be a little cleaner.